src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -283,7 +284,7 @@ private void showCommentLog() {
         uiContext.getMapData().getMapDimensions().height);
     final Image small = uiContext.getMapImage().getSmallMapImage();
     smallView = new MapPanelSmallView(small, model);
-    mapPanel = new MapPanel(data, smallView, uiContext, model);
+    mapPanel = new MapPanel(data, smallView, uiContext, model, this);
```

Comment:
I think there may be a disconnect or confusion. The concern I have is that TripleAFrame references MapPanel, and now MapPanel has a reference back to TripleAFrame. this is a circular dependency. My suggestion to avoid that is to pass MapPanel a function that accesses TripleAFrame. This way only the function reference is passed down to MapPanel, and not the whole TripleAFrame. In addition, TripleAFrame gets to control how it is accessed by defining the function, MapPanel winds up never knowing about TripleAFrame.